### ISSUE TITLE
Оптимизировать верстку ландшафтного режима в мобильном виде

### ISSUE THREAD

**Issue by DJ-ZX** [rekcuFniarB/forum-theprodigy-ru#84]
идеи и скриншот - TBD

**Comment by rekcuFniarB** [comment]
Никакой специальной оптимизации не требуется. На смартфонах всё ок (я вообще пользуюсь только ландшафтной ориентацией всегда, портретная для меня противоестественно). На планшетах там кому как, кому не норм, можно просто выключить мобильный режим, благо размер экрана позволяет и десктопным пользоваться.

Не планшет, но одна фигня:

![screenshot](http://i.imgur.com/GJNKnNO.png "screenshot")
![screenshot](http://i.imgur.com/0nz6vFT.png "screenshot")
![screenshot](http://i.imgur.com/gSVIqja.png "screenshot")
![screenshot](http://i.imgur.com/p3iVVCQ.png "screenshot")

**Comment by DJ-ZX** [comment]
на компе, кстати получше, но не на мобилах, смотри:
![screenshot_20170602-132537](https://cloud.githubusercontent.com/assets/13614247/26753056/be9c332a-4866-11e7-8a6b-232b8c9694e4.png)
![screenshot_20170602-132707](https://cloud.githubusercontent.com/assets/13614247/26753055/be996e60-4866-11e7-9719-13b0171229a3.png)
![wp_ss_20170603_0003](https://cloud.githubusercontent.com/assets/13614247/26753067/e4f8dee2-4866-11e7-8f3a-11735a2bd4d0.png)

сравни с десктоп видом
![wp_ss_20170603_0004](https://cloud.githubusercontent.com/assets/13614247/26753066/e4f7ba08-4866-11e7-8bc5-41dc859dd40f.png)
![desktop_wp_ss_20170603_0001](https://cloud.githubusercontent.com/assets/13614247/26753071/ebb9d470-4866-11e7-82b8-60e53df06f7f.png)
![desktop_wp_ss_20170603_0002](https://cloud.githubusercontent.com/assets/13614247/26753070/eb9f4402-4866-11e7-862b-9af50bd96a44.png)

**Comment by DJ-ZX** [comment]
Если по списку:
1 - шапка занимает полстраницы, зачем её зумить? фактически профит от мобильного вида и скрытия части элементов нивелируется (сравни с десктоп скринами внизу
2 - аналогично в профиле (там бага, напишу отдельно), но десктоп вид особенно в ландшафте выглядит тупо збс, на фоне мобильного (вообще вот профиль можно как раз в мобильном виде оставить как в десктопном, там всё ок, поменять только часть боковой панели
3 - строчка про комментарии вместо того, чтобы растягиваться в одну линию (места много) переносится на вторую или иногда третью строку (если есть комменты и к своим и к чужим и личка, то вроде даже 4 строки), что в портретном вполне ок, а в ландшафтном полная шляпа
4 - в темах пустые или мелкие афиши или элементы вверху страниц занимают опять же очень дофига места, при этом поле поиска рагульное и если в портретном норм, то в ландшафтном это занимает всё неадекватно много места

**Comment by rekcuFniarB** [comment]
> 1 - шапка занимает полстраницы, зачем её зумить? фактически профит от мобильного вида и скрытия части элементов нивелируется (сравни с десктоп скринами внизу

Лого как бы пофиг, всё равно же прокручиваешь вниз. На смартфоне-то оно ещё норм, а на более крупном экране да, уже вырвиглазненко, но если лого не растягивать во всю ширину, то как-то выглятит это не очень тоже. Я б не заморачивался насчёт этого, всё равно все прокручивают вниз, наверху делать нечего в любом случае :D 

> поменять только часть боковой панели

Боковая панель в мобильном режиме и так не отображается по умолчанию, а только по запросу, причём при отображении боковой панели она отображается вместо основного контента профиля до перезагрузки или скрытия этой панели. Хз о чём ты, короче, там уже всё оптимизировано давно.

3) ничего не понял о чём речь.

> поле поиска рагульное 

Что это значит? :D

> если в портретном норм, то в ландшафтном это занимает всё неадекватно много места

Вообще никакой разницы, в высоту занимают место одинаково.

**Comment by DJ-ZX** [comment]
я потом поставлю две моих мобилы рядом и сделаю сравнительные фотки в разных режимах, чтобы пояснить
3 - я про это 
![image](https://cloud.githubusercontent.com/assets/13614247/26756041/cf0b9d18-48a2-11e7-85ba-b3a7cdca2490.png)
2 строчки при ширине куда можно 10 строк вместить, если там будет больше уведомлений, то будет ваще ад

прокручивать дно, лого надо меньше, джын был неправ про убрать ваще, но и пол-экрана это лишнее
кроме того, новые мобилы выходят сча 5.7+ дюймов и ещё и формат 2 к 1, а не 16 на 9, у меня на лопатах и 6 и 6.4 и 6.8 дюймов всё ужасно, а если соотношение вытянуть, то ваще шляпа

аналогичный момент про портретный, когда всё ввысь вытянуто, то потеря 1/6 или 1/5 экрана пофиг, потеря полэкрана оч заметна

**Comment by DJ-ZX** [comment]
Все еще актуально, кстати

**Comment by DJ-ZX** [comment]
не, сие песец и ннада фиксеть )))

**Comment by rekcuFniarB** [comment]
В чём пиздец-то? Норм.

**Comment by DJ-ZX** [comment]
Серьезно?  70% экрана занимают шапка и меню, действительно, ноу проблем

**Comment by DJ-ZX** [comment]
![Image](https://i.imgur.com/3RWhPkL.jpg)
![Image](https://i.imgur.com/O5VFUrv.jpg)
![Image](https://i.imgur.com/ekzVfhZ.jpg)

Просто сравни шапки форумов адоуб и мс
Причем при входе в темы шапки вообще скрываются
Это адище натуральное

**Comment by rekcuFniarB** [comment]
Так у MS лого проще.  А тут то же что и у нас:

![screenshot](https://camo.githubusercontent.com/64e3555e710067265a73508a364f0891ed614d43/68747470733a2f2f692e696d6775722e636f6d2f4f3556465572762e6a7067)

**Comment by DJ-ZX** [comment]
Ну сравни мои скрины, нигде лого на старт странице не занимает более половины экрана и много где при входе в темы и разделы его убирают или еще больше уменьшают. У адоуб лого с меню форума и если вычесть меню всего корпоративного сайта занимает менее 40% страницы и они вмещают туда сайд панель и контент. 
Насчет размеров - а нахера у нас картинка на всю ширину распидорашивается то ваще?

**Comment by DJ-ZX** [comment]
Не, оно в ландшафтном режиме именно растягивается. Картинка становится мыльной. Кроме того шапка не должна расширяться по вертикали, то есть картинку по вертикали нужно вписывать. 
И нет, неважно как в портретном, лого не является главной частью форума, я бы его ваще показывал только гостям в мобильном виде. Насчет недефолтных скинов - мы там по статсам смотрели, те, что юзаются норм надо поддерживать, амбуланс один из них

**Comment by rekcuFniarB** [comment]
> Не, оно в ландшафтном режиме именно растягивается.

Естественно.

> Картинка становится мыльной.

Картинка такая, реальное разрешение картинки ниже разрешения твоего экрана. Я хз зачем ретины на мобилах, оно многие картинки будет замыливать, ибо нафиг не надо из-за яблопользователей утяжелять все картинки на сайтах. (более высокого разрешения лого всё равно нет)

> Кроме того шапка не должна расширяться по вертикали, то есть картинку по вертикали нужно вписывать.

Бред. У изображения есть свои пропорции, если их нарушать, то будет смешно.

![screenshot](https://i.imgur.com/TllWJwm.png)

>  я бы его ваще показывал только гостям в мобильном виде.

В CSS нет таких условий гость - не гость. Когда вынесем вёрстку отдельно от логики (о чём я как раз упоминал в последних комментах в чате), можно будет для мобильного вида использовать отдельные темплейты. Сейчас всё делается только посредством CSS. Иначе это ещё большее засирание кода, который и так адский.

> Насчет недефолтных скинов - мы там по статсам смотрели, те, что юзаются норм надо поддерживать, амбуланс один из них

Амбуланс местами кривой скин. На форуме только два качественно написанных скина, это prodigy и dig7er. Остальное писалось дилетантами.

>  мы там по статсам смотрели,

Они не показывают реальных пользователей. Когда-то Диг практиковал прописывание новых скинов по дефолту чтобы новые скины попиарить. В результате чего дефолтный скин прописывался автоматически новым регистрантам. То есть статистика в основном показывает мертвые души. А кто из живых пользователей (тех кто заходит регулярно на форум), их никак не посмотреть автоматическим запросом.

Алсо, лого в Arenabulance тоже скорей всего выше чем в оригинальном Ambulance.

**Comment by rekcuFniarB** [comment]
Ну и если лого уменьшить, в дефолтном скине будет вот так убого выглядеть:

![screenshot](https://i.imgur.com/kVkmjyD.png)

А в Arenabulance будет не так убого, за счёт того что там фон изображения сливается с фоном страницы.

**Comment by DJ-ZX** [comment]
по последнему скрину как раз то, что я говорил - цсс в ландшафтном должен возвращать блок "сообщения и время" влево, как в десктопном, не думаю, что это нереально

по картинкам - официально взять и всем сделать одинаковую высоту до дефолтного скина одновременно, а потом ширину залить цветом фона, сам поиграешься или скинешь мне все картинки архивом и я приведу к виду, дальше - при любом ракскладе может версия выше и выглядит убого, но картинка хотя бы не распидорашивается на 1900 пикселей или сколько там у айфона, можно сделать хотя бы отступы и в принципе  будет ок, например сча отступов на процентов 45, сделать отступы по 12-15% с краёв строго для ландшафта, я думаю детектить ты можешь, но если нет - пусть будет везде, 12-15% в портрете будет менее заметно, а бесить будет сильно меньше
можешь попробовать и дать на тест

**Comment by DJ-ZX** [comment]
по скинам неважно - кривой или нет, ты можешь попробовать взять статистику активных пользователей зареганых за год (по активности тупо - заходил не позже года назад) и посмотреть какие у кого НЕДЕФОЛТНЫЕ (то есть человек спецом поменял) недефолтные скины, думаю из базы это реально

**Comment by rekcuFniarB** [comment]
>  цсс в ландшафтном должен возвращать блок "сообщения и время" влево, как в десктопном, не думаю, что это нереально

Не годится.
![screenshot](https://i.imgur.com/QUlvzTm.jpg)

> по картинкам - официально взять и всем сделать одинаковую высоту до дефолтного скина

Там везде примерно одинаково, исключение Arenabulance, у которого высота выше остальных. Я не буду что-то там мудрить с этим скином, в этом нет смысла.

Проблема с лого сильно надумана. В любом случае я этим не собираюсь заниматься в ближайшее время.

**Comment by DJ-ZX** [comment]
ну кому надумана, а у кого глазик дёргается
а можно вот в те кастомные настройки добавить "Не отображать лого" и делов то?

**Comment by rekcuFniarB** [comment]
А просто там же сменить скин на другой на данном устройстве нельзя, где лого не такое высокое (почти любой скин)?

Я не вижу смысла городить лишний костыль из-за того что в какой-то кастомном скине высокое лого и кто-то зачем-то пялится на лого и жалуется на него, вместо того чтобы прокрутить дальше. :F

**Comment by DJ-ZX** [comment]
но ваще моё предложение - для ландшафта оставить так
![image](https://user-images.githubusercontent.com/13614247/48739056-185d2b80-ec53-11e8-8b2f-da1079247d2c.png)
если это реально, это не костыль, избавит от распидорашивания и всё будет ок, как на реальных мониторах

**Comment by rekcuFniarB** [comment]
Нет, это в ширину не влезает на смартфонах, особенно когда есть уведомления о комментах. А на планшетах в ландшафте мобильный вид и вовсе не нужен.

> я сделаю аренабуланс нормальной высоты и ширины без растягивания, могу и остальным сделать

С Аренабуланс делай что хочешь, а остальное ломать не надо, там всё оптимально.

**Comment by DJ-ZX** [comment]
блин, брейн, чо ты несёшь? это скрин со смартфона выше, причём 16:9, могу и с фуллхд сделать
сча тебе с 18:9 сделаю с мобилы

**Comment by rekcuFniarB** [comment]
Какое это имеет значение? С чего ты взял что верстать нужно обязательно под самые широкие лопаты?

**Comment by DJ-ZX** [comment]
какие самые широкие? все экраны в смартах 100 лет как 16:9, телефоны меньше 5.5 почти не выпускаются, считай 5.5 стандарт уже года 2 (кроме айфонов на 4.7 и то - с прошлого года не выпускаются уже), все новые телефоны выпускаются 16:9
ты в каком мире живёшь то?

**Comment by DJ-ZX** [comment]
![Image](https://i.imgur.com/iYbqASg.jpg)
![Image](https://i.imgur.com/RZd8pPW.jpg)
![Image](https://i.imgur.com/hLnO1z7.jpg)
![Image](https://i.imgur.com/tGIf97U.jpg)
![Image](https://i.imgur.com/fzKxFo5.jpg)

**Comment by DJ-ZX** [comment]
вот тебе айфон хс макс (айфон х-хс тоже самое, поверь), выше в посте с люмии, могу сделать с нокии на 4.3 дюйма скрины - надо?

**Comment by rekcuFniarB** [comment]
А что ты хотел этими скринами сказать? Они говорят только о том что мобильный режим тут даже и не нужен: https://camo.githubusercontent.com/f0c81449efe935a17dc7ad4edd34166bb24a73c4/68747470733a2f2f692e696d6775722e636f6d2f684c6e4f317a372e6a7067

**Comment by DJ-ZX** [comment]
нет, он не нужен только в шапке, в разделах и списках тем уже не всё так красиво, как в шапке (ну и в профиле, как видишь)

кроме того чтобы ты понимал - у меня стандартные шрифты и зум экрана везде в минимум, если я поставлю стандарное, то ваще будет срака и на винфоне и на андроиде и на яфоне
то есть зумируй всё что видишь процентов на 15-20

**Comment by rekcuFniarB** [comment]
А зачем ты так долго смотришь на лого?

**Comment by DJ-ZX** [comment]
я бы не смотрел, а так открываю форум - а там тока лого

**Comment by rekcuFniarB** [comment]
Так прокрути. Это фишка лого Arenabulance, усмехающиеся над посетителем головы. 

Не вижу каких-то прям проблем вселенского масштаба с этим. Лого занимает где-то 30% экрана, дальше идёт контент и очевидно что надо прокручивать дальше. 

![scr](https://i.imgur.com/l1uDWoL.jpg)
![scr](https://i.imgur.com/R0LN4Rm.jpg)
![scr](https://i.imgur.com/8nbXCWJ.jpg)

(что-то у меня при снятии скриншота часть верхней шторки попадает на снимок)

В любом случае, отображать уведомления и лого на одной строке не буду, как я уже сказал, это годится только для каких-то крупныл лопат, но ломает всё остальное. Текущий вариант оптимальный.

**Comment by DJ-ZX** [comment]
вот ты мало что упрямый, так и игноришь, что я не только про лого

**Comment by rekcuFniarB** [comment]
А про что ещё? Ты больше про лого говоришь. Страница длинная, всё на первый экран не влезет всё равно.  Меню видно, уведомления видно, то что дальше есть контент очевидно.
